src/App.js

```diff
@@ -53,6 +54,32 @@ class App extends Component {
         }
     }
 
+	/**
+	 * This function will take a text input and checks if it is valid json
+	 * @param {String} jsonFile The json file entered by the user
+	 * @returns {Boolean} The determination if the file is valid json
+	 */
+	validateJSON(jsonFile) {
+		//first a quick way to check the size of the file is to check the character length
+		//Let's assume first of all that a JSON input file should be no larger than 1MB, or 1048576 characters
+		if(jsonFile.length > 1048576) {
+			return false;
+		}
+		//Second check if it is a valid JSON with JSON.parse 
+		var validJson;
+		try {
+			validJson = JSON.parse(jsonFile);
+		} catch(e) {
+			return false; 
+		}
+		//finally check if there are more than 1000 items in the json array or 1000 keys in the object
+		if(Object.keys(validJson).length > 1000) {
+			return false;
```

Comment:
We might need to show feedback that only 1000 items are allowed in array

Reply:
That's a good idea

Reply:
lets put that as validation error in the same error area

Reply:
What you could do is have the validateJSON return back either an object with the error details, or an error string. In case of both, if no errors return null.

Reply:
yeah, something like `error: {message: "", el: "element to highlight"}`